docs/addons/instana-addon.md

```diff
@@ -0,0 +1,198 @@
+2. [kubectl](https://Kubernetes.io/docs/tasks/tools/)
+3. [cdk](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_install)
+4. [npm](https://docs.npmjs.com/cli/v8/commands/npm-install)
+5. Instana backend application - Use SaaS (eg [aws](https://aws.amazon.com/marketplace/pp/prodview-hnqy5e3t3fzda?sr=0-1&ref_=beagle&applicationId=AWSMPContessa)) or Install self-hosted Instana backend ([on-premises](https://www.ibm.com/docs/en/instana-observability/current?topic=installing-configuring-self-hosted-instana-backend-premises))
+
+## Installation
+
+To create a new project and install dependencies, follow these steps from Amazon EKS Blueprint [Quick Start](https://aws-quickstart.github.io/cdk-eks-blueprints/getting-started)
+
+Use following command to Install IBM Instana Addon:
+
+```shell
+npm i @instana/aws-eks-blueprint-addon
+```
+
+## Instana Agent Configuration
+Go to your Instana installation (Instana User Interface), click ... More > Agents > Installing Instana Agents and select 'Kubernetes' platform to get the Instana Agent Key, Instana Service Endpoint, Instana Service port. These steps are also described [here](https://www.ibm.com/docs/en/instana-observability/218?topic=instana-endpoints-keys) or in the screenshot below.
+
+![Instana Agent Configuration](../assets/images//instana-agent.png)
+
+To set the following environment variables from the CLI, use the corresponding values obtained from the Instana Service Endpoint and Port (as shown in the above screenshot), and the Instana Application Key (also shown in the above screenshot):
+
+- Set the value of **INSTANA_ENDPOINT_HOST_URL** to the Instana Service Endpoint.
+- Set the value of **INSTANA_ENDPOINT_HOST_PORT** to the Instana Service Port.
+- Set the value of **INSTANA_AGENT_KEY** to the Instana Application Key.
+
+You can choose the names for **AMAZON_EKS_CLUSTER_NAME** and **INSTANA_ZONE_NAME** based on your cluster's name, for example, "eks-blueprint".
+
+Set the value of the following environment variable and run it on CLI to set those variables.
+
+```
+export INSTANA_ZONE_NAME=
+export AMAZON_EKS_CLUSTER_NAME=
+export INSTANA_AGENT_KEY=
+export INSTANA_ENDPOINT_HOST_URL=
+export INSTANA_ENDPOINT_HOST_PORT=
+```
+
+For example:
+
+```
+export INSTANA_ZONE_NAME=eks-blueprint
+export AMAZON_EKS_CLUSTER_NAME=eks-blueprint
+export INSTANA_AGENT_KEY=abc123
+export INSTANA_ENDPOINT_HOST_URL=instana.example.com
+export INSTANA_ENDPOINT_HOST_PORT="443"
+```
+
+## Usage
+```typescript
+import * as blueprints from "@aws-quickstart/eks-blueprints";
+import { loadYaml } from "@aws-quickstart/eks-blueprints/dist/utils";
+import * as cdk from "aws-cdk-lib";
+import { InstanaOperatorAddon } from "@instana/aws-eks-blueprint-addon";
+
+const instanaProps = {
+  zone: {
+    name: process.env.INSTANA_ZONE_NAME, // Mandatory Parameter
+  },
+  cluster: {
+    name: process.env.AMAZON_EKS_CLUSTER_NAME, // Mandatory Parameter
```

Comment:
Why do we need to collect the EKS cluster name from the user? It will lead to ambiguity. Let's say the cluster name is "dev-cluster" but they put "prod-cluster" in value. 
It seems it would be more prudent to just default this to the actual cluster name or something like `${region}-$clusterName}`. 
You can always get the cluster name from ClusterInfo which is passed to the deploy method of your add-on: ` clusterInfo.cluster.clusterName`.

Reply:
Implemented usage of AWS Secrets Manager secrets.